docs/user-guide/builtins.md

```diff
@@ -116,6 +116,74 @@ Such named modules will always then be available under the
     Please see the documentation (linked above) about restrictions and gotchas
     when configuring how JavaScript modules are made available to PyScript.
 
+### `pyscript.fetch`
+
+A common task is to `fetch` data from the web via HTTP requests. The
+`pyscript.fetch` function provides a uniform way to achieve this in both
+Pyodide and MicroPython. It is closely modelled on the
+[Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) found
+in browsers with some important Pythonic differences.
+
+The simple use case is to pass in a URL and `await` the response. Remember, in
+order to use `await` you must have the `async` attribute in the `script` tag
+that references your code.
+
+```python title="A simple HTTP GET with pyscript.fetch"
+from pyscript import fetch
+
+
+response = await fetch("https://example.com")
+```
+
+If the `fetch` operation _returns a response that is not deemed `OK`_ (the
+definition of which
+[can be found here](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok))
+then an exception is raised.
+
+Assuming an `OK` response, the following methods are available to you to access
+the data returned from the server:
+
```

Comment:
these are **all asynchronous** ... we removed the double await need but it's needed for any of these variants ... the `fetch` does pretty much nothing until you state what kind of result you are after ... once again:

  * await fetch(url).text()
  * await fetch(url).json()
  * await fetch(url).bytearray()
  * await fetch(url).arrayBuffer()
  * await fetch(url).blob()

That's it, don't split this in 2 operations or we offer nothing more than native `fetch` then.

Reply:
P.S. eventually you can say you can:

```python
from pyscript import fetch

remote = fetch(url)

await remote.text()
# and all others
```

The target goal for the fetch is the most important and async thing we need and there's no workaround for that to be different, it's how every other *fetch* works out there, including the pyodide thing that requires double await ... I hope this is clear now, a bit surprised it wasn't already.